src/ui/marker.js

```diff
@@ -512,6 +535,10 @@ export default class Marker extends Evented {
                 DOM.setTransform(this._element, `${anchorTranslate[this._anchor]} translate(${this._pos.x}px, ${this._pos.y}px) ${pitch} ${rotation}`);
             }
         });
+
+        if ((this._map.getTerrain() || this._map.getFog()) && !this._fadeTimer) {
+            this._fadeTimer = setTimeout(this._evaluateOpacity.bind(this, true), 60);
```

Comment:
Is this throttling here because querying fog opacity is too expensive to do on every frame or some other reason?

It adds a bunch of complexity here that would be avoided if we could update the opacity together with the position

Reply:
> This isn't a concept newly introduced by this PR

Yep, you're right. Let's leave as is

> Can you be more clear about what "bunch of complexity" could be removed immediately?

Small things like the bug [this](https://github.com/mapbox/mapbox-gl-js/pull/10564/commits/5e436a564da2c072453e1b403b755d9488d0227b) may have fixed which are simple on their own but easily slip into anything dealing with timeouts. A duplicate unproject. Not showing up as part of a render call tree in profiling. The method is sometimes called within a render frame and sometimes not. Updating different properties of the dom element in different places at different times. None of these things are huge

Reply:
Makes sense, captured this as a follow up to evaluate removal of throttling.